sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java

```diff
@@ -426,9 +408,10 @@ private long getEstimatedSizeBytes(
         }
 
         LOG.debug("Number of splits is {}", splitKeys.size());
-        for (String shardFilter : splitKeysToFilters(splitKeys, spec.filter())) {
-          sources.add(new BoundedMongoDbSource(spec.withFilter(shardFilter)));
-        }
+        // TODO: What should be done here?
+        // for (String shardFilter : splitKeysToFilters(splitKeys, spec.filter())) {
```

Comment:
It seems `spec.filter()` is pretty ineffective here. Let me take care of this part because I want to validate the way splitting is working.

Reply:
Now that you mentioned this. In a project I'm using internally I faced issues with splitting with a MongoDB cluster hosted by Atlas. The MongoDB version I'm using is 3.4. I will share the error/exception thrown by MongoDB shortly for more context on this issue.

Reply:
Now that you mention it, there is an open ticket of the connector not working in atlas, if you are up to check this one and eventually fix it, please self assign the ticket [BEAM-4567](https://issues.apache.org/jira/browse/BEAM-4567). It will be a really welcomed fix, please tell me if you are interested and I will do the review / help you.

Reply:
I’ll take a look and let you know.